### PR TITLE
Make IP address range list required when embargoing.

### PR DESCRIPTION
**Pull request by JojoVes**
**JIRA Ticket**: [ISLANDORA-2032](https://jira.duraspace.org/browse/ISLANDORA-2032)

# What does this Pull Request do?

Makes things not break when someone tries to embargo an item without an actual IP range list.

# What's new?
Makes the IP range list required, and makes it impossible to submit a null value.

# How should this be tested?

To reproduce the problem, try using IP Embargo to embargo an object without choosing an IP range list. Screenshot of the error is attached to the JIRA ticket.

This pull request should stop the form submission before it gets to the errors by making the field required.

# Interested parties
@Islandora/7-x-1-x-committers

### CODE REVIEW

File: includes/forms.inc
Comment (qadan) [inline includes/forms.inc:451]: The 'none' list item is [intended to be used below](https://github.com/Islandora-Labs/islandora_ip_embargo/blob/6761ba3eb9930c229a43dffdbee8f3b576898182/includes/forms.inc#L520) as a way of removing IP embargo lists when updating ... seems like the problem is more the fact that you can try to create a new list using a list ID of 'none' and it just happily tries to add that to the database.

My recommendation would be:
* Give the 'none' list option an actual label (e.g., `array('none' => t('No IP Embargo'))` so it's much more clear what it's trying to do
* Remove the `#required` designation
* Create an `islandora_ip_embargo_object_embargo_form_validate()` handler that ensures `$form_state['values']['lists']` is not 'none' when `$form_state['storage']['islandroa_ip_embargo']['update'] == TRUE`

**Comment by DiegoPino** [comment]
@JojoVes @qadan could we avoid showing No IP Embargo option at all for objects that don´t have embargoes applied instead of adding messaging to go around an option that won´t do anything other than making users ask why the option is there in the first place? Seems like better UX approach to hide not needed options. In that case, would be also a better option to have a "remove IP embargo" option for those that have and need the embargo to be removed since that is what is triggered? The whole wording, logic seems over-complicatedly being exposed to a final user.

**Comment by willtp87** [comment]
@DiegoPino Those would be good UI changes, but are out of scope. This is a fix to current functionality.  Feel free to open up a subsequent ticket.

**Comment by DiegoPino** [comment]
@willtp87 since we are so few islandora committers reviewing code, it could be beneficial if tickets could grow into a wider scope so a full solution (or at least one that survives longer) could be applied. There is danger in partial fixes that don't address the full scope, especially on supporting those via documentation. Said that, if you all feel this is enough then I can make another ticket. But: this one needs to be upgraded to "Bug" not "Code Tasks". That error message is, in fact, a bug.

**Comment by JojoVes** [comment]
@DiegoPino As much as I agree with the UI needing to be fixed since I didn't even know about this way of removing the embargo before working on this, it does make sense to separate UI into a different ticket so this fix can make it through faster.

I tried changing the 'type' to 'Bug' instead of 'Task' on the ticket, and it won't let me. It only gives the options of 'Task' or 'Code Task'.

**Comment by DiegoPino** [comment]
@JojoVes seems like @willtp87 got it updated to Bug.
Ok, will go for another ticket on the UX stuff, giving people the option to remove something that is not there is just weird. Thanks

File: includes/forms.inc
Comment (DiegoPino) [inline includes/forms.inc]: Hey @JojoVes 😬  i will approve and merge, just wondering if this (real code task!) could get done here also?
Says: " form_set_error('islandora_ip_embargo_overlay_text_color', t('Color must match hexideciamal format, requires 3 or 6 characters and A-F 0-9 characters apply (E.g. F0A3B1 or F1A).'));"
kinda could be "hexadecimal". Just a tought, if you happen to still have the file open...

**Comment by DiegoPino** [comment]
@JojoVes added a little comment there, real low hanging fruit. Maybe can be added as low hanging fruit here? https://github.com/Islandora-Labs/islandora_ip_embargo/pull/34/files#r133280246 If not will merge in 24 hours anyway. looks good.

**Comment by willtp87** [comment]
@DiegoPino  This merge is not under a 24h clock.

**Comment by DiegoPino** [comment]
@willtp87 true! Will start by Approving? =)

**Review by DiegoPino — APPROVED** [review]

**Pull request merged**
